Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/internal/triangle_soup_snap_rounding.h

```diff
@@ -0,0 +1,585 @@
+    CGAL::Polygon_mesh_processing::internal::remove_invalid_polygons_in_polygon_soup(points, polygons);
+    CGAL::Polygon_mesh_processing::merge_duplicate_polygons_in_polygon_soup(points, polygons, np);
+    CGAL::Polygon_mesh_processing::remove_isolated_points_in_polygon_soup(points, polygons);
+  }
+};
+
+// Specialization for array and Indexes_range of array
+template <typename PointRange, typename PolygonRange, typename PID, std::size_t N>
+struct Triangle_soup_fixer<PointRange, PolygonRange, std::array<PID, N> >
+{
+  template <typename NamedParameters>
+  void operator()(PointRange& points,
+                  PolygonRange& polygons,
+                  const NamedParameters& np) const
+  {
+    repair_polygon_soup(points, polygons, np);
+  }
+};
+
+template <typename PointRange, typename PolygonRange, typename PID, std::size_t N>
+struct Triangle_soup_fixer<PointRange, PolygonRange, Indexes_range< std::array<PID, N> > >
+{
+  template <typename NamedParameters>
+  void operator()(PointRange& points,
+                  PolygonRange& polygons,
+                  const NamedParameters& np) const
+  {
+    using parameters::get_parameter;
+    using parameters::choose_parameter;
+
+    typedef typename GetPolygonGeomTraits<PointRange, PolygonRange, NamedParameters>::type Traits;
+    Traits traits = choose_parameter(get_parameter(np, internal_np::geom_traits), Traits());
+
+    CGAL::Polygon_mesh_processing::merge_duplicate_points_in_polygon_soup(points, polygons, np);
+    CGAL::Polygon_mesh_processing::internal::remove_invalid_polygons_in_array_polygon_soup(points, polygons, traits);
+    CGAL::Polygon_mesh_processing::merge_duplicate_polygons_in_polygon_soup(points, polygons, np);
+    CGAL::Polygon_mesh_processing::remove_isolated_points_in_polygon_soup(points, polygons);
+  }
+};
+
+template <typename PointRange, typename PolygonRange, typename NamedParameters>
+void repair_triangle_soup(PointRange& points,
+                          PolygonRange& polygons,
+                          const NamedParameters& np)
+{
+  Triangle_soup_fixer<PointRange, PolygonRange> fixer;
+  fixer(points, polygons, np);
+}
+
+// A visitor of Autorefinement to track the correspondance between input and output triangles
+struct Wrap_snap_visitor : public Autorefinement::Default_visitor
+{
+  template< typename Triangle>
+  void new_subdivision(Triangle& new_t, const Triangle& old_t) {
+    new_t.set_id(old_t.id());
+  }
+};
+
+/**
+*
+* Rounds the coordinates of the points so that they fit in doubles while making and keeping the model intersection free by potentially subdividing the triangles.
+* The input can be any triangle soup and the output is an intersection-free triangle soup with Hausdorff distance
+* between the input and the output bounded by M*2^-gs*k where M is the maximum absolute coordinate in the model, gs the snap_grid_size (see below) and k the number of iteration
```

Comment:
```suggestion
* between the input and the output bounded by `M*2^-gs*k` where `M`is the maximum absolute coordinate in the model, `gs` the `snap_grid_size` (see below) and `k` the number of iterations
```

Please check if the superscript `^` produces what you expect.